webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ScoreService.scala

```diff
@@ -158,9 +162,16 @@ class ScoreServiceImpl(
   }
   def cleanScore(name: String)(implicit qc: QueryContext): IOResult[Unit] = {
     for {
-      _         <- cache.update(_.map { case (id, gscore) => (id, gscore.copy(details = gscore.details.filterNot(_.scoreId == name))) })
+      _         <- ScoreLoggerPure.info(s"Cleaning score ${name}")
+      // Remove from score caches
+      newGlobal <- cache.updateAndGet(_.map {
+                     case (id, gscore) => (id, gscore.copy(details = gscore.details.filterNot(_.scoreId == name)))
+                   })
       newScores <- scoreCache.updateAndGet(_.map { case (id, scores) => (id, scores.filterNot(_.scoreId == name)) })
-      _         <- update(newScores)
+      // Remove from databases
+      _         <- ZIO.foreach(newScores.keySet) { case nodeId => scoreRepository.deleteScore(nodeId, Some(name)) }
```

Comment:
I think we want to have a repo method with signature like `def deleteScore(name: ScoreId, nodes: Seq[NodeId]): IOResult[???]` so that we can do only one sql request with a prepared statement and avoid high I/O when there is a lot of nodes involved.